homeassistant/components/rpi_gpio/binary_sensor.py

```diff
@@ -62,12 +63,16 @@ def __init__(self, name, port, pull_mode, bouncetime, invert_logic):
 
         rpi_gpio.setup_input(self._port, self._pull_mode)
 
-        def read_gpio(port):
+        def read_gpio():
             """Read state from GPIO."""
             self._state = rpi_gpio.read_input(self._port)
             self.schedule_update_ha_state()
 
-        rpi_gpio.edge_detect(self._port, read_gpio, self._bouncetime)
+        def edge_detected(port):
+            """Edge detection handler."""
+            threading.Timer(float(self._bouncetime) / 1000, read_gpio).start()
```

Comment:
Let's not use a thread but instead schedule it on the event loop.

```suggestion
		async def async_read_gpio(self):
		    """Read state from GPIO."""
		    await asyncio.sleep(float(self._bouncetime) / 1000)
		    self._state = await self.hass.async_add_executor_job(rpio_gpio.read_input, self._port)
		    self.async_write_ha_state()

        def edge_detected(port):
            """Edge detection handler."""
            self.hass.add_job(self.async_read_gpio)
```